app/src/main/java/org/fcitx/fcitx5/android/input/candidates/CandidateItemUi.kt

```diff
@@ -34,4 +41,48 @@ class CandidateItemUi(override val ctx: Context, theme: Theme) : Ui {
             gravity = gravityCenter
         })
     }
+
+    private var promptMenu: PopupMenu? = null
+
+    fun showExtraActionMenu(currentIm: InputMethodEntry, onForget: () -> Unit) {
+        promptMenu?.dismiss()
+        val context = root.context
+        promptMenu = PopupMenu(context, root).apply {
+            val actions = arrayListOf<Pair<String, () -> Unit>>().apply {
+                // only Pinyin Engine has forget word feature
+                if (currentIm.uniqueName == "pinyin" || currentIm.uniqueName == "shuangpin") {
```

Comment:
`table` based input methods also have the shortcut to "delete a word" in Ctrl+7, it's better to benfit them as well. I would add `addon` field to `InputMethodEntry` class later.